### PR TITLE
boards: Adding board definition for rpi_pico_w

### PR DESCRIPTION
**Pull request by ghost**
This is a barebones implementation of the board definition for the Raspberry Pi
Pico W. Effort has been made to abstract the features that are common to both
the Pico and Pico W into a shared file when possible. This commit does not
include the addition of bluetooth or wifi drivers for the W's Infineon
module.

### CODE REVIEW

**Comment by beechwoods-software** [comment]
Sorry, didn't mean to create this PR (yet).

**Comment by beechwoods-software** [comment]
Thanks for the pointer. I hadn’t noticed the “multiple board revisions” section in the documentation yet (I’d really just modeled this after the two variants of the “teensy” board). 

I hadn’t quite meant to submit this as a PR yet this morning (I’d just intended to sync up with the main branch and clicked the wrong button), but will do so soon.

—Dave

**Comment by beechwoods-software** [comment]
> @beechwoods-software
> 
> We're welcome your contributions. At first glance, it looks good in general. IMO, if you plan to commonize the setting with original pico, it is good to consider using "board revision".
> 
> https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#multiple-board-revisions

Hi, 

I just tried using the "board revision" in my local build tree to differentiate between the original pico and the pico W.

It works, but I'm concerned that "board revision" isn't really the right way to handle different families of boards like the non-wireless and wireless version of the Pico. It seems that "board revision" is really intended for serial versioning of hardware (i.e. where each version of a board is denoted by an incremental number or letter). 

In the case of the current pico, it would require me to choose a version code for the non-wireless Pico. I could just choose an arbitrary letter like "A" and then call the wireless version "W". I could then make "A" the default so that people could still build the original pico with "west build -p -b rpi_pico" and people who want to build the wireless version could probably figure out that they need to do "west build -p -b rpi_pico@W". That would make sense for now, but what would happen when new revisions of each of these boards come out? My understanding is that you must either choose a numbering or a lettering version scheme (you can't combine the two schemes), so for future versions of the Pico I think there would be no choice other than to start naming them after arbitrary letters (maybe "B" and "X"?). 

Am I missing something here? Is there a way to combine numbering and lettering in the "board revision" style of versioning? It seems like a lot of the other families of boards that I see in the current Zephyr codebase just defined entirely new names for different categories of boards that are in the same family of boards like I did in the changeset that is associated with this PR.

**Comment by yonsch** [comment]
@beechwoods-software Maybe a better approach would be having a common base board that the two boards inherit from. An example of this is the `efr32_radio_brd*` family which inherits from `efr32_radio`

**Comment by beechwoods-software** [comment]
> @beechwoods-software Maybe a better approach would be having a common base board that the two boards inherit from. An example of this is the `efr32_radio_brd*` family which inherits from `efr32_radio`

Thanks. I'll look at that example.

**Comment by beechwoods-software** [comment]
OK, I think what I had at first is pretty similar to how efr32_radio did this (although I'd originally used the teensy variants as a model). I'm going to re-open this is a submission with a few additional changes pushed (I abstracted the common parts of the dts into a dtsi and updated documentation).

File: boards/arm/rpi_pico/rpi_pico_w.dts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2021 Yonatan Schachter
```
Comment (soburi) [inline boards/arm/rpi_pico/rpi_pico_w.dts:2]: Please write your name in the new file.

**Review by soburi — CHANGES_REQUESTED** [review]
Firstly, please clean up your commit log with `rebase -i`.
(Simply says, remove merge commits.)
And you need to update the description of this PR.

If PR contains only one commit, you can use the same description with the commit message. If not so, please summarize what changes with these commits.

File: boards/arm/rpi_pico/rpi_pico_w.dts (hunk 1 above)
Comment (beechwoods-software) [reply boards/arm/rpi_pico/rpi_pico_w.dts:2]: Sorry, I'm not quite sure ow to use "rebase -i" to remove the merge commits. I re-synced the beechwoods-software fork with your repo, then ran 'git rebase -i' from a workspace where I have the branch that I'm submitting, but the merge commits are still there. Did I miss an instruction?

Is there something specific you need in the description for the PR, or do you just want some more detail for the purpose of the PR?

There is more than one non-merge commit, but they each have separate commit messages that describe the purpose of the commits, so I think everything should be good there.

Comment (beechwoods-software) [reply boards/arm/rpi_pico/rpi_pico_w.dts:2]: @soburi Do you have any pointers for how to remove the merge commits? A Google search is not very helpful in this regard, because a lot of the information that I see is somewhat contradictory and most of it sounds somewhat dangerous.

**Comment by beechwoods-software** [comment]
OK, I think I have the issues with the merge comments resolved and I've updated the PR description and also put my name on the one new file.

**Review by soburi — CHANGES_REQUESTED** [review]

File: boards/arm/rpi_pico/rpi_pico_w.dts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2023 Dave Rensberger - Beechwoods Software
```
Comment (soburi) [inline boards/arm/rpi_pico/rpi_pico_w.dts:2]: Combine first commit and last commit by squash.

**Review by soburi — CHANGES_REQUESTED** [review]

File: boards/arm/rpi_pico/rpi_pico_w.dts (hunk 1 above)
Comment (soburi) [reply boards/arm/rpi_pico/rpi_pico_w.dts]: > OK, I think I have the issues with the merge comments resolved and I've updated the PR description and also put my name on the one new file.

It correctly removed the merge commits. Thank you.

Commit messages still not follows guidelines.
Please confirm the following.

https://docs.zephyrproject.org/3.1.0/contribute/guidelines.html#commit-guidelines

Commit message will be following format.

```
[category]: [sub-category]: [summary]

[description...]
[description...]
[description...]

Signed-off-by: Your Name <your.email@example.com>
```

File: boards/arm/rpi_pico/rpi_pico_w.dts (hunk 2 above)
Comment (beechwoods-software) [reply boards/arm/rpi_pico/rpi_pico_w.dts:2]: I assume that you wouldn't have any problem with my squashing _all_ of the comments into one for my next push?

File: boards/arm/rpi_pico/rpi_pico.dts [hunk 3]
```diff
@@ -14,119 +14,4 @@
 
 #include <zephyr/dt-bindings/i2c/i2c.h>
```
[the remark this answers is not in the document]
Comment (beechwoods-software) [reply boards/arm/rpi_pico/rpi_pico.dts]: Good idea. Done.

File: boards/arm/rpi_pico/rpi_pico_w.dts (hunk 1 above)
Comment (beechwoods-software) [reply boards/arm/rpi_pico/rpi_pico_w.dts]: I believe my commit now conforms to this format.

File: boards/arm/rpi_pico/rpi_pico_w.dts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2021 Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
[the remark this answers is not in the document]
Comment (beechwoods-software) [reply boards/arm/rpi_pico/rpi_pico_w.dts:6]: I hadn't noticed that. That part of the device tree is only in rpi_pico.dts now.

File: boards/arm/rpi_pico/rpi_pico_w.dts (hunk 2 above)
Comment (beechwoods-software) [reply boards/arm/rpi_pico/rpi_pico_w.dts:2]: I combined all commits into a single summary commit now.

File: boards/arm/rpi_pico/rpi_pico_w.dts (hunk 4 above)
[the remark this answers is not in the document]
Comment (soburi) [reply boards/arm/rpi_pico/rpi_pico_w.dts:6]: Sorry, it my misunderstanding.

File: boards/arm/rpi_pico/Kconfig.defconfig [hunk 5]
```diff
@@ -1,10 +1,11 @@
 # Copyright (c) 2021 Yonatan Schachter
 # SPDX-License-Identifier: Apache-2.0
 
-if BOARD_RPI_PICO
+if BOARD_RPI_PICO || BOARD_RPI_PICO_W
 
 config BOARD
-	default "rpi_pico"
+	default "rpi_pico" if BOARD_RPI_PICO
+	default "rpi_pico_w" if BOARD_RPI_PICO_W
```
Comment (soburi) [inline boards/arm/rpi_pico/Kconfig.defconfig:8]: There is an unwanted trailing white space.

**Review by soburi — CHANGES_REQUESTED** [review]

File: boards/arm/rpi_pico/doc/index.rst [hunk 6]
```diff
@@ -28,6 +29,7 @@ Hardware
 - 8 Programmable I/O (PIO) for custom peripherals
 - On-board LED
 - 1 Watchdog timer peripheral
+- Infineon CYW43439 2.4 GHz Wi-Fi chip (Pico W only)
 
 
 .. figure:: img/rpi_pico.jpg
```
Comment (soburi) [inline boards/arm/rpi_pico/doc/index.rst:35]: You don't need an image if you don't reference it from index.rst.

**Comment by soburi** [comment]
offtopic: Please don't "Resolve conversation" button. Reviewers usually use it to remember "What I pointed out".

**Comment by beechwoods-software** [comment]
Sorry, didn't mean to click on "please re-review" yet.

File: boards/arm/rpi_pico/doc/index.rst (hunk 6 above)
Comment (beechwoods-software) [reply boards/arm/rpi_pico/doc/index.rst:35]: Added a reference.

File: boards/arm/rpi_pico/Kconfig.defconfig (hunk 5 above)
Comment (beechwoods-software) [reply boards/arm/rpi_pico/Kconfig.defconfig]: Removed the extra whitespace.

**Comment by beechwoods-software** [comment]
> offtopic: Please don't "Resolve conversation" button. Reviewers usually use it to remember "What I pointed out".

Sorry, I thought that was to acknowledge that I'd acted on the comment. I won't do it anymore.

**Comment by beechwoods-software** [comment]
New review comments addressed.

**Comment by beechwoods-software** [comment]
Reduced image size for the Pico W (and also cropped it to look more like the image of the plain Pico).

**Comment by beechwoods-software** [comment]
Could someone help point me to how I would go about reproducing just the problem with the adc_api unit test? Running "scripts/twister -c -p rpi_pico -p rpi_pico_w" reproduces the problem, but it runs /everything/, which takes hours.

**Comment by beechwoods-software** [comment]
The push I just made fixes the adc_api unit test problem. It appears that when you create a new board name for a board that's very similar to an existing board, any instance of a dts overlay in the build tree for the existing board needs to be have a <new_board_name>.overlay file created. 

This fixes the problem in the same way that everyone else seems to have done it, but something seems wrong about this pattern. In most cases, this new file is just a copy of the one for the existing board with a different name. I used symlinks instead, because that seemed slightly better than a verbatim copy, but this still seems less than ideal. Is there any effort underway to try to come up with a more ideal way to handle this?

File: tests/drivers/adc/adc_api/boards/rpi_pico_w.overlay [hunk 7]
```diff
@@ -0,0 +1 @@
+rpi_pico.overlay
```
Comment (soburi) [inline tests/drivers/adc/adc_api/boards/rpi_pico_w.overlay:1]: We're not usually create link in repository.
Please create copy of rpi_pico.overlay.
(Also, Don't forget rewrite copyright by your name.)

**Review by soburi — CHANGES_REQUESTED** [review]

File: tests/drivers/adc/adc_api/boards/rpi_pico_w.overlay (hunk 7 above)
Comment (beechwoods-software) [reply tests/drivers/adc/adc_api/boards/rpi_pico_w.overlay]: Done.

**Review by soburi — CHANGES_REQUESTED** [review]
Great job!
It's almost here.

@yonsch 
Could you take look at?

File: boards/arm/rpi_pico/doc/index.rst [hunk 8]
```diff
@@ -28,14 +29,22 @@ Hardware
 - 8 Programmable I/O (PIO) for custom peripherals
 - On-board LED
 - 1 Watchdog timer peripheral
+- Infineon CYW43439 2.4 GHz Wi-Fi chip (Pico W only)
 
-
+
```
[the remark this answers is not in the document]
Comment (beechwoods-software) [reply boards/arm/rpi_pico/doc/index.rst]: Done.

File: boards/arm/rpi_pico/doc/index.rst [hunk 9]
```diff
@@ -6,11 +6,12 @@ Raspberry Pi Pico
...
 ********
 
-The Raspberry Pi Pico is a small, low-cost, versatile board from
-Raspberry Pi. It is equipped with an RP2040 SoC, an on-board LED,
-a USB connector, and an SWD interface. The USB bootloader allows it
-to be flashed without any adapter, in a drag-and-drop manner.
-It is also possible to flash and debug the Pico with its SWD interface,
+The Raspberry Pi Pico and Pico W are small, low-cost, versatile boards from
+Raspberry Pi. They are equipped with an RP2040 SoC, an on-board LED,
+a USB connector, and an SWD interface. The Pico W additionally conatains an
```
Comment (yonsch) [inline boards/arm/rpi_pico/doc/index.rst:11]: "conatains" is a typo

File: boards/arm/rpi_pico/doc/index.rst [hunk 10]
```diff
@@ -85,13 +94,22 @@ hardware features:
      - :kconfig:option:`CONFIG_SERIAL`
      - :dtcompatible:`raspberrypi,pico-uart-pio`
 
+The rpi_pico_w board configuration will additionally support WiFi and
```
Comment (yonsch) [inline boards/arm/rpi_pico/doc/index.rst:97]: I don't think those lines are necessary as the wifi module is not listed in the supported features

File: boards/arm/rpi_pico/doc/index.rst [hunk 11]
```diff
@@ -85,13 +94,22 @@ hardware features:
...
 Pin Mapping
 ===========
 
 The peripherals of the RP2040 SoC can be routed to various pins on the board.
 The configuration of these routes can be modified through DTS. Please refer to
 the datasheet to see the possible routings for each peripheral.
 
+External pin mapping on the Pico W is identical to the Pico, but note that internal
+RP2040 GPIO lines 23, 24, 25, and 29 are routed to the Infineon module on the W.
+Since GPIO 25 is rounted to the on-board LED on the Pico, but to the Infineon module
```
Comment (yonsch) [inline boards/arm/rpi_pico/doc/index.rst:109]: "rounted" is a typo

**Review by yonsch — CHANGES_REQUESTED** [review]
Looks good generally, added a few nitpicks

File: boards/arm/rpi_pico/doc/index.rst (hunk 11 above)
Comment (beechwoods-software) [reply boards/arm/rpi_pico/doc/index.rst]: Thanks. All 3 changes made as requested and pushed.

**Comment by yonsch** [comment]
Great, thanks. One last thing: the image you're using is pretty blurry. Can you use an image from the official documentation? But make sure the resolution is not too high as this increases the repository's size

**Comment by beechwoods-software** [comment]
> Great, thanks. One last thing: the image you're using is pretty blurry. Can you use an image from the official documentation? But make sure the resolution is not too high as this increases the repository's size

Can you point me to the image that you used for the original pico? I was only able to find an image with all 4 pico variants (non-wireless and wireless, with and without headers) on the raspberrypi foundation website. I tried to crop just the pico w out of that image, but I can't seem to get the size to match the one in the rpi_pico.jpg image.

**Comment by beechwoods-software** [comment]
OK, I think I got the image size right now.

**Comment by soburi** [comment]
Please add descriptions continue to header line  with one blank line on commit message. (See `git log` result)
And copy to PR comment that message.

**Comment by beechwoods-software** [comment]
> Please add descriptions continue to header line with one blank line on commit message. (See `git log` result) And copy to PR comment that message.

I'm sorry, I don't understand what you're asking me to do here.

**Comment by soburi** [comment]
> I'm sorry, I don't understand what you're asking me to do here.

Commit message usually take following format.

```
boards: arm: rpi_pico: [header line]

[descrption]

Signed-off-by: <your@mail.address>
```

In this case, need to add [description] part.

```
boards: rpi_pico: The barebones changes for a board called rpi_pico_w 
```

I think more simple title are prefer.
For example, "Add rpi_pico_w board".
And the details write to [description] part.

**Comment by beechwoods-software** [comment]
> @beechwoods-software
> 
> > I'm sorry, I don't understand what you're asking me to do here.
> 
> Commit message usually use following format.
> 
> ```
> boards: arm: rpi_pico: [header line]
> 
> [descrption]
> 
> Signed-off-by: <your@mail.address>
> ```
> 
> In this case, need to add [description] part.
> 
> ```
> boards: rpi_pico: The barebones changes for a board called rpi_pico_w 
> ```
> 
> I think more simple title are prefer. It is preferable the current title of this PR, "Add board definition for rpi_pico_w". And the details write to [description] part.

OK, Got it now. I think the changes that I just pushed should address those concerns.

**Comment by soburi** [comment]
@beechwoods-software 
Please update PR Message with commit description.
edit-pr-msg

**Comment by beechwoods-software** [comment]
> @beechwoods-software Please update PR Message with commit description. edit-pr-msg

OK, done.

**Review by soburi — APPROVED** [review]

**Review by yonsch — DISMISSED** [review]
Thanks!

**Comment by soburi** [comment]
@beechwoods-software 
> Your automated check didn't like the name on my signoff. I think I just fixed that.

Please check `git config user.name`

**Comment by beechwoods-software** [comment]
> @beechwoods-software
> 
> > Your automated check didn't like the name on my signoff. I think I just fixed that.
> 
> Please check `git config user.name`

It shows the right name:

$ git config user.name
Dave Rensberger

I think the problem is, though, that I initiated the PR from my company's github account ('beechwoods-software'), while the commits and pushes came from my own github account ('Dave Rensberger'). In the future, I'll make sure that members of my team initiate their PRs from their own github accounts rather than the company one, but I'm not sure exactly how to fix the present situation with this PR. It's getting late, so I'll look into it tomorrow.

**Comment by soburi** [comment]
@beechwoods-software 
CI checks equivariant of Author: and Signed-Off.
It's hard to fix Author:, so look at the diff and recreate the commit again.
You must use your private account for commit and sign-off.

CI doesn't check which account push.
You can use your company account.

**Review by soburi — CHANGES_REQUESTED** [review]
Fix CI errors.

**Comment by beechwoods-software** [comment]
Closing this PR, because I need to fix the format of the "Author", which is evidently impossible without creating a new PR.

**Comment by beechwoods-software** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/57739 has been created with exactly the same changeset, description, etc as this one in order to fix the issue with the "Author" field.

**Comment by soburi** [comment]
@beechwoods-software 
Sorry, I misunderstood that #57739 fixed.
It need to recreate the commit.
(rebase cannot change Author: field)
There are some ways.
Most simple way, execute `git reset HEAD~1` to  revert current commit 
and re-add and commit.
Of course, need to re-type commit message.
It will fill the Author: field of commit with current `user.name` and `user.email`.

**Comment by soburi** [comment]
> I'm even more confused now. Should we do this in 57437 or 57439? I agree that it would not be good to lose all of the conversation history on 57437.

I want to keep going and complete #57437, and I think it can.

>  ....and should we create a new branch with the same name or not? It would be ideal for us to keep the branch name the same, but not absolutely necessary.

It is github's system requirement. PR ties to branch name. Need to keep it.

> I'm not feeling comfortable that just deleting the old rpi_pico_w and replacing it with a new branch of the same name isn't going to really screw a lot of things up, but you sound like you have dealt with this before.

Renaming Author can do follwing way.
But this is also somewhat cumbersome.
I find it easier to reset and recommit.
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_changing_email_addresses_globally

**Comment by beechwoods-software** [comment]
Reopening

**Comment by beechwoods-software** [comment]
Reopening

**Comment by soburi** [comment]
Still branch head is d96424f2.
Could you please push with -f option?
And PR status seem `closed`. Is the "Reopen and comment" button active in your view?

**Comment by beechwoods-software** [comment]
I had already tried 'push -f', but it just says "Everything up-to-date" and does nothing, which was why I wondered if I have to do the "reset HEAD~1" and re-patch/re-commit thing again.

**Comment by soburi** [comment]
Please push d96424f as `rpi_pico_w` branch with force option.
It will make able to reopen this PR.
After that, push 9f30610 as `rpi_pico_w`.

References this instructions.

https://gist.github.com/robertpainsi/2c42c15f1ce6dab03a0675348edd4e2c

**Comment by beechwoods-software** [comment]
I can't push https://github.com/zephyrproject-rtos/zephyr/pull/57437#:~:text=Please%20push-,d96424f,-as%20rpi_pico_w%20branch%20with%20force%20option.%0AIt because it no longer exists due to rebasing.

**Comment by soburi** [comment]
Please do following.

```
git branch -D rpi_pico_w 
git fetch origin pull/57437/head:rpi_pico_w
git push -f [your_repository] rpi_pico_w
```

**Comment by beechwoods-software** [comment]
It says "fatal: couldn't find remote ref pull/57437/head" when I do the fetch command.

Should I just make a superficial change to https://github.com/zephyrproject-rtos/zephyr/commit/9f306108b60f20c5cd128307fcdef99bf4dac877 and push that?

**Comment by soburi** [comment]
Confirmation: What display with `git remote -v` in your environment?

To reopen the PR you need to push d96424f to rpi_pico_w.
Once reopened, push 9f30610 to rpi_pico.

**Comment by beechwoods-software** [comment]
OK, I see. I didn't have "remote" set up (only "origin"). Let me try again.

**Comment by beechwoods-software** [comment]
It's reopened now, but I'm not quite sure how to get https://github.com/zephyrproject-rtos/zephyr/commit/9f306108b60f20c5cd128307fcdef99bf4dac877 back.

**Comment by soburi** [comment]
> It's reopened now, but I'm not quite sure how to get https://github.com/zephyrproject-rtos/zephyr/commit/9f306108b60f20c5cd128307fcdef99bf4dac877 back.

Fetch #57739 commit.

git branch -D rpi_pico_w 
git fetch origin pull/57739/head:rpi_pico_w
git push -f [your_repository] rpi_pico_w

**Comment by beechwoods-software** [comment]
> https://gist.github.com/robertpainsi/2c42c15f1ce6dab03a0675348edd4e2c

I cannot figure out how to revive the 9f30610 commit. 

The https://gist.github.com/robertpainsi/2c42c15f1ce6dab03a0675348edd4e2c link that you referred me to is a dead link, by the way.

**Comment by soburi** [comment]
What happen `git log 9f30610` in your local?

**Comment by beechwoods-software** [comment]
OK, I think I'm closer now. The 'git fetch origin pull/57739/head:rpi_pico_w' worked (I had actually tried that much earlier, but I'd gotten the ID of the other PR wrong... I thought it was 57439, not 57739).

I was also able to do the force push, but it doesn't seem to be being seen by the 57437 PR yet.

**Comment by beechwoods-software** [comment]
> OK, I think I'm closer now. The 'git fetch origin pull/57739/head:rpi_pico_w' worked (I had actually tried that much earlier, but I'd gotten the ID of the other PR wrong... I thought it was 57439, not 57739).
> 
> I was also able to do the force push, but it doesn't seem to be being seen by the 57437 PR yet.

Everything else looks like it's good to me though (the PR is associated with the right commit and the commit exists in the repository). I see some "HTTP 500" errors in the automated test logs.... that may have been due to the way that we were messing with things, or it may have been that github was just inaccessible for a brief period (I got some 500s on the command line too).  Is there any way you can force those to run again?

**Comment by soburi** [comment]
> Everything else looks like it's good to me though (the PR is associated with the right commit and the commit exists in the repository). I see some "HTTP 500" errors in the automated test logs.... that may have been due to the way that we were messing with things, or it may have been that github was just inaccessible for a brief period (I got some 500s on the command line too). Is there any way you can force those to run again?

Github seems unstable now.

**Comment by beechwoods-software** [comment]
> > Everything else looks like it's good to me though (the PR is associated with the right commit and the commit exists in the repository). I see some "HTTP 500" errors in the automated test logs.... that may have been due to the way that we were messing with things, or it may have been that github was just inaccessible for a brief period (I got some 500s on the command line too). Is there any way you can force those to run again?
> 
> Github seems unstable now.

I agree. As far as I can tell, though, I think everything should be ready to go for this PR once it comes back. Let me know if you think there's anything else I need to do.

**Comment by soburi** [comment]
> I agree. As far as I can tell, though, I think everything should be ready to go for this PR once it comes back. Let me know if you think there's anything else I need to do.

No more things to do at this time. You only need to push 9f30610.

**Comment by soburi** [comment]
> I agree. As far as I can tell, though, I think everything should be ready to go for this PR once it comes back. Let me know if you think there's anything else I need to do.

Please re-run failed CI jobs.

**Comment by beechwoods-software** [comment]
> > I agree. As far as I can tell, though, I think everything should be ready to go for this PR once it comes back. Let me know if you think there's anything else I need to do.
> 
> Please re-run failed CI jobs.

Do I have the ability to do that? I'm not seeing that option anywhere.

**Comment by beechwoods-software** [comment]
There was still one compliance issue with the commit comment. I pushed that fix and all of the automated tests passed, so hopefully we're finally good to go?

**Comment by soburi** [comment]
@beechwoods-software 
Still remain CI checks. Please fix it.
Message no need to indent.

**Comment by beechwoods-software** [comment]
> @beechwoods-software Still remain CI checks. Please fix it. Message no need to indent.

Is the indentation what it's actually complaining about?  I'll have to look into how to edit long commit messages like that. When I needed to rebase, it let me do it in a text editor, but when I was just fixing the commit message I had to do it on the command line and I guess it just inserted indentations for some reason.

**Comment by soburi** [comment]
> Is the indentation what it's actually complaining about?

Sorry, It is my misjudgment.

Following command to fix message.

```
git rebase -i 7a946450e #in this PR case
```

You will get open editor, change `pick` to `reword` in first line, and edit message.

**Comment by beechwoods-software** [comment]
> > Is the indentation what it's actually complaining about?
> 
> Sorry, It is my misjudgment.
> 
> Following command to fix message.
> 
> ```
> git rebase -i 7a946450e #in this PR case
> ```
> 
> You will get open editor, change `pick` to `reword` in first line, and edit message.

Where did the "7a946450e" come from?

...and also, what do I do after that. I don't understand the state that it left my repository in after I did that command. I tried to do a push and it first said "you're not on a branch", but when I checked out the rpi_pico_w branch, it seemed to undo my commit message edit.

**Comment by soburi** [comment]
> Where did the "7a946450e" come from?

This is the hash of the one before commit from commit that do in this PR.

> ...and also, what do I do after that? I don't understand the state that it left my repository in after I did that command. I tried to do a push and it first said "you're not on a branch", but when I checked out the rpi_pico_w branch, it seemed to undo my commit message edit.

Oh, I forget last command.
Please run

```
git rebase --continue
```

**Comment by beechwoods-software** [comment]
Is the compliance check still complaining about the current push? Does it want the entire comment to be all on one line?

**Comment by soburi** [comment]
https://github.com/zephyrproject-rtos/zephyr/actions/runs/4952442970/jobs/8858802054?pr=57437

Second line must be blank line.

**Comment by beechwoods-software** [comment]
OK, All one line with a blank line after. Just pushed.

**Comment by beechwoods-software** [comment]
> OK, All one line with a blank line after. Just pushed.

Actually I don't think it's going to like that one either. I just reviewed the guidelines again and I think I see how they want it. I'll try again.

**Comment by beechwoods-software** [comment]
Actually, I still don't see how to do this with rebase.  Do I need to put a "reword <changeset>" in front of every line? ...or was I on the right track when I just made it all one line?

**Comment by beechwoods-software** [comment]
OK, it looks kind of funny to me all on one line, but it passed, so I'm absolutely fine with that!

**Comment by soburi** [comment]
>  1: UC5 Commit title exceeds max length (277>75): 

one line must be under 75 characters.

**Comment by beechwoods-software** [comment]
> > 1: UC5 Commit title exceeds max length (277>75):
> 
> one line must be under 75 characters.
> 
> Don't put into one line!
> 
> Please check CI error message and https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines.

If I do multiple lines, then git rebase complains (says "error: invalid line" for every line except the first). I understand the guidelines, I don't understand how to make the comment conform to them using git.

**Comment by beechwoods-software** [comment]
OK, it seems that the problem was that I was changing the message as part of the reword command, whereas I was supposed to wait for the secondary editor (the one in the command is "just for reference" and actually gets ignored). https://stackoverflow.com/questions/50494713/how-can-i-insert-line-breaks-to-reworded-commit-messages-during-git-interactive

Git's user interface truly is an abomination, but it's what the world has chosen. ;)

Hopefully this one works!

**Comment by soburi** [comment]
Great job!
All CI passed.

**Review by soburi — APPROVED** [review]

**Comment by soburi** [comment]
@yonsch 
Could you take a look?

**Comment by beechwoods-software** [comment]
> Great job! All CI passed.

Thanks for all your help. I'll try to help make sure that my Beechwoods colleagues make sure to do the initial PR from an account that has the "Author" set up correctly, because my experience here is that one is rather painful to correct.

**Review by yonsch — APPROVED** [review]

**Comment by soburi** [comment]
> Thanks for all your help. I'll try to help make sure that my Beechwoods colleagues make sure to do the initial PR from an account that has the "Author" set up correctly, because my experience here is that one is rather painful to correct.

Great! It is giant step for human beings! 👍
